include/bluetooth/audio/has.h

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) 2021 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_BLUETOOTH_SERVICES_HAS_H_
+#define ZEPHYR_INCLUDE_BLUETOOTH_SERVICES_HAS_H_
+
+/**
+ * @brief Hearing Access Service (HAS)
+ *
+ * @defgroup bt_has Hearing Access Service (HAS)
+ *
+ * @ingroup bluetooth
+ * @{
+ *
+ * The Hearing Access Service is used to identify a hearing aid and optionally to control hearing
+ * aid presets. This API implements the server functionality.
+ *
+ * [Experimental] Users should note that the APIs can change as a part of ongoing development.
+ */
+
+#include <zephyr/types.h>
+#include <bluetooth/bluetooth.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @brief The preset record contains all the information about a hearing aid preset. */
+struct bt_has_preset {
+	/** @brief Preset set active callback
+	 *
+	 *  The callback can also be used locally in which case no connection will be set.
+	 *  Once the preset has changed, the bt_has_active_preset_changed stall be called
+	 *  to complete the procedure.
+	 *
+	 *  @param conn   The connection that is requesting to read
+	 *  @param preset The preset that's being set active
+	 *  @param sync   Whether the server must relay this change to the other member of
+	 * 				  the Binaural Hearing Aid Set.
```

Comment:
Is that really the responsibility of the HA server to do? Isn't it the HARCs responsibility to sync multiple HAs?

Reply:
HA can do it if supports Preset Synchronization feature

Reply:
This is one of the questions that I have always asked in the HAWG, still waiting for a response. 

To me it doesn't make sense that a HA (with HAS) should be responsible for relaying information to the other coordinated set members, when the procedure is also defined for the HARC. 

The only use case I can imagine is if the user can change this on the HA itself, and there's no HARC around.

Reply:
But there is no requirement that the information shall be relayed over HARC. This can be done other vendor-specific way. Note that the HA on the market already communicate with each other without BT

Reply:
> But there is no requirement that the information shall be relayed over HARC.

That depends. e.g. for the name write:

> The HARC shall execute this procedure with both HAs that are part of a Binaural Hearing Aid Set, if the Independent Presets flag is set to 0b0, and the HARC shall use the same parameter values during both executions.

or the set active preset:

> The HARC shall execute this procedure with both HAs that are part of a Binaural Hearing Aid Set, if the Independent Presets flag is set to 0b0, and the HARC shall use the same parameter values during both executions.

The "Synchronized Locally procedure" are the only ones that are being communicated between the two HA without HARC. Given that CAP requires the use of CSIP for e.g. changing the Volume, I don't see the point of having the locally synchronized procedures at all, since the HARC will likely be connected to both earbuds in any case. 

But this is more of a comment for the HAWG rather than this PR, so I'll wait for their response.